### PR TITLE
Feature/temper with connection clean up

### PR DESCRIPTION
**Pull request by NikoBergemann**
### Changes in this PR
- remove obsolete code
- clean up Code snippets
- add missing new Proposal Handling for admission Connection
- add Test to check chaincode status after successfull approval
- add Test to check approval ledger state
- add Execution of "real" transaction upon final approval submit 
- change UnsignedTransactionTests to use Matriculation-Transaction instead of CertificateAction to actually change ledgerData

### CODE REVIEW

File: src/main/scala/de/upb/cs/uc4/hyperledger/connections/traits/ConnectionTrait.scala [hunk 1]
```diff
@@ -118,56 +117,43 @@ trait ConnectionTrait extends AutoCloseable {
...
+
+    // TODO: test parse params to execute real transaction
+    //approvalResult
+    val realResult = internalSubmitRealTransactionFromApprovalProposal(params)
+    new Gson().toJson(Seq(approvalResult, realResult))
   }
 
-  def internalSubmitRealTransactionFromApprovalProposal(proposal: Proposal): String = {
-    val (proposalContractName, transactionName, params) = TransactionHelper.getParametersFromApprovalProposal(proposal)
+  def internalSubmitRealTransactionFromApprovalProposal(params: Seq[String]): String = {
```
Comment (anemonekk) [inline src/main/scala/de/upb/cs/uc4/hyperledger/connections/traits/ConnectionTrait.scala:138]: change to protected

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/helper/TransactionHelper.scala [hunk 2]
```diff
@@ -236,12 +183,12 @@ protected[hyperledger] object TransactionHelper {
       transactionPayload: ByteString
   ): CompletableFuture[BlockEvent#TransactionEvent] = {
     try {
-      val orderers = if (ReflectionHelper.getPrivateField(transactionOptions)("orderers")() != null) ReflectionHelper.getPrivateField(transactionOptions)("orderers")().asInstanceOf[util.List[Orderer]]
-      else new util.ArrayList[Orderer](channel.getOrderers())
+      val orderers = if (ReflectionHelper.getPrivateField(transactionOptions)("orderers") != null) ReflectionHelper.getPrivateField(transactionOptions)("orderers").asInstanceOf[util.List[Orderer]]
+      else new util.ArrayList[Orderer](channel.getOrderers)
       val shuffeledOrderers: util.ArrayList[Orderer] = new util.ArrayList[Orderer](orderers)
-      if (ReflectionHelper.getPrivateField(transactionOptions)("shuffleOrders")().asInstanceOf[Boolean]) Collections.shuffle(shuffeledOrderers)
+      if (ReflectionHelper.getPrivateField(transactionOptions)("shuffleOrders").asInstanceOf[Boolean]) Collections.shuffle(shuffeledOrderers)
```
Comment (anemonekk) [inline src/main/scala/de/upb/cs/uc4/hyperledger/utilities/helper/TransactionHelper.scala:189]: rename string to shuffleOrderers

**Review by anemonekk — APPROVED** [review]

**Review by matthias-geuchen — APPROVED** [review]
Looks good.

File: src/main/scala/de/upb/cs/uc4/hyperledger/connections/traits/ConnectionTrait.scala (hunk 1 above)
Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/connections/traits/ConnectionTrait.scala:138]:
```suggestion
  private def internalSubmitRealTransactionFromApprovalProposal(params: Seq[String]): String = {

```

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/helper/TransactionHelper.scala (hunk 2 above)
Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/helper/TransactionHelper.scala:189]: I think shuffelOrders is the correct Field, is it not?

**Review by mco33415 — APPROVED** [review]

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/helper/TransactionHelper.scala (hunk 2 above)
Comment (anemonekk) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/helper/TransactionHelper.scala:189]: Just a minor thing, the field is called shuffledOrderers, so plus "er" in the end, the string is missing that

**Review by sluis1 — APPROVED** [review]

File: src/main/scala/de/upb/cs/uc4/hyperledger/utilities/helper/TransactionHelper.scala (hunk 2 above)
Comment (NikoBergemann) [reply src/main/scala/de/upb/cs/uc4/hyperledger/utilities/helper/TransactionHelper.scala:189]:
```java
/**
     * TransactionOptions class can be used to change how the SDK processes the Transaction.
     */
    public static class TransactionOptions {
        List<Orderer> orderers;
        boolean shuffleOrders = true;
        NOfEvents nOfEvents;
        User userContext;
        boolean failFast = true;

....


        public TransactionOptions shuffleOrders(boolean shuffleOrders) {
            this.shuffleOrders = shuffleOrders;
            return this;
        }
```
this is the transactionOptions class of the transactionOptions-object we want to access the field "shuffelOrders" not the method "shuffleOrderers"

**Pull request merged**